BAZProject/SearchMovie/SearchMovieViewController.swift

```diff
@@ -10,12 +10,76 @@
 
 import UIKit
 
-class SearchMovieViewController: UIViewController, SearchMovieViewProtocol {
+final class SearchMovieViewController: UIViewController, SearchMovieView {
 
-	var presenter: SearchMoviePresenterProtocol?
+	var presenter: SearchMoviePresenter?
+    @IBOutlet weak var txtSearch: UISearchBar!
+    @IBOutlet weak var movieCollectionView: UICollectionView!
+    private var movies: [MovieModel] = [] {
+        didSet {
+            DispatchQueue.main.async { [weak self] in
+                self?.movieCollectionView.reloadData()
+            }
+        }
+    }
 
-	override func viewDidLoad() {
+    override func viewDidLoad() {
         super.viewDidLoad()
+        configureCollectionView()
+        txtSearch.delegate = self
+    }
+
+    private func configureCollectionView() {
+        let layout = UICollectionViewFlowLayout()
+        layout.itemSize = CGSize(width: 120, height: 120)
+        movieCollectionView.collectionViewLayout = layout
+        movieCollectionView.register(SearchMovieViewCell.nib(), forCellWithReuseIdentifier: "SearchMovieViewCell")
+        movieCollectionView.delegate = self
+        movieCollectionView.dataSource = self
     }
 
+    private func getMoviesToCollection(query: String) {
+        presenter?.getMovies(endPoint: SearchEndPoint.init(query: query), endPointResult: { [weak self] resultEndPoint in
+            if resultEndPoint != nil {
+                if let result = resultEndPoint {
+                    self?.movies = result.movies
+                }
+            } else {
+                return
+            }
+        })
+    }
 }
+
+extension SearchMovieViewController: UICollectionViewDelegate, UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        movies.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "SearchMovieViewCell", for: indexPath) as? SearchMovieViewCell else {
+            return UICollectionViewCell()
+        }
+        if let poster = movies[indexPath.row].moviePoster {
+            cell.configureCell(with: poster)
+        }
+        return cell
+    }
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return CGSize(width: 120, height: 120)
+    }
+
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        let detailView = MovieDetailRouter.createModule(movie: movies[indexPath.row])
+        self.navigationController?.pushViewController(detailView, animated: true)
+    }
```

Comment:
Los Routers deberían servirte para manejar esta navegación, esa es la intención del Router en VIPER.
Intenta en el futuro manejar esa navegación desde el router. Lo puede hacer usando el patrón `Delegate` y manteniendo una referencia al navigationController desde el router.

También te recomiendo la lectura del [Coordinator Pattern](https://www.hackingwithswift.com/articles/71/how-to-use-the-coordinator-pattern-in-ios-apps) por si te interesa investigar más al respecto de cómo desacoplar la nevgación del viewcontroller.